python/README.md

```diff
@@ -0,0 +1,187 @@
+## Getting started
+This module contains the building blocks to build new Flow connectors in Python, as well as adapting external Python connectors written in standard protocols to Flow. A Python connector is a subclass of the `Connector` classes in the `capture`, `derive`, and `materialize` (once it's written) submodules. Convenience classes `shim_airbyte_cdk.CaptureShim` and `shim_singer_sdk.CaptureShim` are provided to adapt existing Airbyte or Singer/Meltano connectors.
+
+A Python connector is composed of a few critical pieces:
+* `pyproject.toml` defines the Python module and its dependencies. For example:
+  ```toml
+  [tool.poetry.dependencies]
+  # If we're pulling in an open-source connector that doesn't need any changes, we can define it here. Subdirectory can be omitted if the connector lives at the root of the repo.
+  source_bigdata = { git = "https://github.com/big/data.git", subdirectory = "connector" }
+  # If we're working with an open-source connector that we have pulled in-tree, we can define it like this, where the connector was pulled into the subdirectory "upstream-connector"
+  source_bigdata = { path = "upstream-connector" }
+  ```
+* In addition to being useful documentation for other people running your connector, `test.flow.yaml` or some other valid Flow spec is needed to invoke your connector locally. You'll point `flowctl` at this file to test your connector when developing locally.
+* `__main__.py` is the entrypoint of your connector. In the case that your connector is using a shim to wrap a supported open-source connector protocol, the file will be quite simple:
+  ```python
+  from python import shim_airbyte_cdk
+  from source_bigdata import SourceBigData
+
+  shim_airbyte_cdk.CaptureShim(
+      delegate=SourceBigData(),
+      oauth2=None
+  ).main()
+  ``` 
+* A `Dockerfile` to build your connector into a container image that will be run in production. An example dockerfile can be found in this directory. Make sure to `source-changeme` with the name of your connector. If you are writing a materialization connector, make sure to also change the `LABEL FLOW_RUNTIME_PROTOCOL` to `materialization`.
+
+## Pulling an open-source connector in tree
+While using existing builds of open-source connectors is all well and good, there will come a time in every connector's life that we need to make changes to it. Before deciding to pull a connector in-tree, first make sure that the version of the connector you're going to pull is licensed to allow this usage. Then, the connector can be imported like so
+
+```bash
+$ git remote add -f --no-tags bigdata https://github.com/big/data.git 
+$ ./python/pull_upstream.sh bigdata/master path/in/source/repo path/in/this/repo
```

Comment:
nit: maybe make these examples more explicit, like:

./python/pull_upstream.sh bigdata/master upstream/path/to/source-foobar ./source-foobar

Also, is there a way to pass in a specific revision?

Reply:
There is now!